appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/visitor/OpenApiWalker.java

```diff
@@ -117,25 +117,7 @@
 
     public OpenApiWalker(OpenAPI api, Set<Class<?>> allowedClasses) {
         this.api = api;
-        this.classes = new TreeSet<>((class1, class2) -> {
-            if (class1.equals(class2)) {
-                return 0;
-            }
-            // Subclasses first
-            if (class1.isAssignableFrom(class2)) {
-                return -1;
-            }
-            // Non contextual objects at the start
-            if (!class1.isAnnotationPresent(ApplicationPath.class) && !class1.isAnnotationPresent(Path.class)) {
-                return -1;
-            }
-            // Followed by applications
-            if (class1.isAnnotationPresent(ApplicationPath.class)) {
-                return -1;
-            }
-            // Followed by everything else
-            return 1;
-        });
```

Comment:
I think @MattGill98 knows about the logic here and if the order had any real significance other than performance.

Reply:
Looks like a legitimate hangover from when the schema searching was less sophisticated. I don't think this serves much of a purpose any longer. As the TCK passes, our tests pass and my manual testing seems all okay, I'm happy with this change